tests/integration_tests/test_model_downloader_udf_implementation.py

```diff
@@ -0,0 +1,75 @@
+from pathlib import Path
+from typing import Dict
+from exasol_bucketfs_utils_python import list_files, delete
+from exasol_transformers_extension.udfs.model_downloader_udf import \
+    ModelDownloader
+from tests.utils.parameters import bucketfs_params
+
+
+class Connection:
+    def __init__(self, address: str, user: str = None, password: str = None):
+        self.address = address
+        self.user = user
+        self.password = password
+
+
+class ExaEnvironment:
+    def __init__(self, connections: Dict[str, Connection] = None):
+        self._connections = connections
+        if self._connections is None:
+            self._connections = {}
+
+    def get_connection(self, name: str) -> Connection:
+        return self._connections[name]
+
+
+class Context:
+    def __init__(self,
+                 model_name: str,
+                 bfs_conn: str):
+        self.model_name = model_name
+        self.bfs_conn = bfs_conn
+        self._emitted = []
+
+    def emit(self, *args):
+        self._emitted.append(args)
+
+    def get_emitted(self):
+        return self._emitted
+
+
+def test_model_downloader_udf_implementation(setup_database, bucket_config):
+    bucketfs_conn_name = "bucketfs_connection"
+    model_name = 'bert-base-uncased'
+    model_path = model_name.replace("-", "_")
+
+    ctx = Context(
+        model_name,
+        bucketfs_conn_name
+    )
+
+    bucketfs_connection = Connection(
```

Comment:
I would replace the real bucketfs with the Mock, see BucketFSLocationFactory, because then you can run this without the DB and as unit test

Reply:
agree :+1: